args/core/async/readonly_rw_spinlock.hpp

```diff
@@ -324,35 +324,51 @@ namespace args::core::async
 	 *		 Read-only operations will only wait for Read-Write operations to be finished.
 	 * @ref args::core::async::readonly_rw_spinlock
 	 */
+	template<size_type S>
 	class readonly_multiguard
 	{
 	private:
-		std::vector<readonly_rw_spinlock*> m_locks;
+		std::array<readonly_rw_spinlock*, S> m_locks;
+
+		template<size_type I, typename... types>
+		void fill(readonly_rw_spinlock& lock, types&&... args)
+		{
+			if constexpr (I > 1)
+			{
+				fill<I - 1>(args...);
+			}
+
+			m_locks[I - 1] = &lock;
+		}
 
 	public:
 		/**@brief Creates readonly multi-guard and locks for Read-only.
 		 */
 		template<typename lock_type1 = readonly_rw_spinlock, typename lock_type2 = readonly_rw_spinlock, typename... lock_typesN>
 		readonly_multiguard(lock_type1& lock1, lock_type2& lock2, lock_typesN&... locks)
 		{
-			m_locks.push_back(&lock1);
-			m_locks.push_back(&lock2);
-			(m_locks.push_back(&locks), ...);
+			fill<sizeof...(locks) + 2>(lock1, lock2, locks...);
```

Comment:
why even use fill here ?
```suggestion
			m_locks = {{ &lock1, &lock2, &locks... }};
```
longer example with better optimization as well https://godbolt.org/z/rWx7vx

Reply:
got it